File: 0/0/9/009.c
// Copyright (C) 2013 Jorge Aparicio

#include <stdio.h>

#define PERIMETER 1000
#define HALF_PERIMETER PERIMETER / 2

int main() {
  int a, b, c;

  for (c = HALF_PERIMETER; c > 0; c--) {
    for (b = c - 1; b > 0; b--) {
      a = PERIMETER - b - c;

      if (a*a + b*b == c*c) {
        printf("%d\n", a * b * c);

        return 0;
      }
    }
  }
}

File: 0/0/9/009.hs
-- Copyright (C) 2013 Jorge Aparicio

main :: IO()
main
  = print
  $ head
  [ a * b * c
  | c <- [halfPerimeter,halfPerimeter-1..1]
  , b <- [c-1,c-2..1]
  , let a = perimeter - b - c
  , a*a + b*b == c*c
  ] where perimeter = 1000 :: Int
          halfPerimeter = perimeter `div` 2
